### ISSUE TITLE
`apple_static_library` fails to cross-compile macOS library for arm64

### ISSUE THREAD

**Issue by lelandjansen** [bazelbuild/bazel#14373]
### Description

`apple_static_library` fails to cross-compile a macOS library for arm64. See the analysis below for details.

This issue does not exist when creating an `apple_static_library` for iOS (i.e., `--ios_multi_cpus=arm64,x86_64` works).

#### Files

```c++
// lib.h
#ifndef LIB_H_
#define LIB_H_

int Add(int a, int b);

#endif  // LIB_H_
```

```c++
// lib.cc
#include "lib.h"

int Add(const int a, const int b) { return a + b; }
```

```bazel
# BUILD
cc_library(
    name = "lib",
    srcs = ["lib.cc"],
    hdrs = ["lib.h"],
)

apple_static_library(
    name = "lib_macos",
    platform_type = "macos",
    deps = [":lib"],
)
```

```bazel
# WORKSPACE
# empty
```

#### Build command

```bash
bazel build //:lib_macos --macos_cpus=arm64  # Building on x86_64
```

```
INFO: Build option --macos_cpus has changed, discarding analysis cache.
INFO: Analyzed target //:lib_macos (0 packages loaded, 335 targets configured).
INFO: Found 1 target...
ERROR: /Users/ljansen/Desktop/example/BUILD:10:21: Linking lib_macos-fl.a failed: (Exit 1): libtool failed: error executing command external/local_config_cc/libtool @bazel-out/macos-arm64-applebin_macos-darwin_arm64-fastbuild/bin/lib_macos-fl.a-2.params

Use --sandbox_debug to see verbose messages from the sandbox
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: no library created (no object files in input files matching -arch_only x86_64)
Target //:lib_macos failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.297s, Critical Path: 0.06s
INFO: 2 processes: 2 internal.
FAILED: Build did NOT complete successfully
```

### Analysis

Even though we're cross-compiling for arm64, `-arch_only` is still set as x86_64 when compiling the `apple_static_library`. `libtool` is unable to find an x86_64 library because the `cc_library` was (correctly) cross-compiled for arm64.

```bash
cat bazel-out/macos-arm64-applebin_macos-darwin_arm64-fastbuild/bin/lib_macos-fl.a-2.params
```

```
-D
-no_warning_for_no_symbols
-static
-arch_only
x86_64     <-------------------------------------------------------------------
-syslibroot
__BAZEL_XCODE_SDKROOT__
-o
bazel-out/macos-arm64-applebin_macos-darwin_arm64-fastbuild/bin/lib_macos-fl.a
bazel-out/macos-arm64-applebin_macos-darwin_arm64-fastbuild/bin/liblib.a
```

```bash
lipo -info bazel-out/macos-arm64-applebin_macos-darwin_arm64-fastbuild/bin/liblib.a
```

```
Non-fat file: bazel-out/macos-arm64-applebin_macos-darwin_arm64-fastbuild/bin/liblib.a is architecture: arm64
```

### Operating system

macOS Big Sur 11.6.1 (20G224)

### Host hardware

2.4 GHz 8-Core Intel Core i9 (x86_64)

### Xcode toolchain

```bash
xcodebuild -version
```

```
Xcode 13.1
Build version 13A1030d
```

### `bazel info release`

`release 4.2.1-homebrew`

**Comment by keith** [comment]
Can you test with bazel 5.0.0rc2? This repro case appears to be fixed for me there, many related fixes have been made

**Comment by lelandjansen** [comment]
Tested and confirmed the fix in `bazel-5.0.0-pre.20211011.2-darwin-x86_64`. Thanks, @keith.
